trade-mapping-to-ipaffs/src/main/java/uk/gov/defra/stw/mapping/toipaffs/chedd/CheddPurposeMapper.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.defra.stw.mapping.toipaffs.chedd;
+
+import static uk.gov.defra.stw.mapping.toipaffs.enumeration.Purpose.TRANSHIPMENT;
+import static uk.gov.defra.tracesx.notificationschema.representation.enumeration.PurposeGroupEnum.TRANSHIPMENT_TO;
+
+import java.util.List;
+import org.springframework.stereotype.Component;
+import uk.gov.defra.stw.mapping.dto.SpsAuthenticationType;
+import uk.gov.defra.stw.mapping.dto.SpsCertificate;
+import uk.gov.defra.stw.mapping.toipaffs.Mapper;
+import uk.gov.defra.stw.mapping.toipaffs.common.PurposeMapper;
+import uk.gov.defra.stw.mapping.toipaffs.exceptions.NotificationMapperException;
+import uk.gov.defra.tracesx.notificationschema.representation.Purpose;
+
+@Component
+public class CheddPurposeMapper implements Mapper<SpsCertificate, Purpose> {
+  private final PurposeMapper purposeMapper;
+
+  public CheddPurposeMapper(PurposeMapper purposeMapper) {
+    this.purposeMapper = purposeMapper;
+  }
+
+  @Override
+  public Purpose map(SpsCertificate spsCertificate) throws NotificationMapperException {
+    List<SpsAuthenticationType> spsAuthenticationTypeList = spsCertificate.getSpsExchangedDocument()
+        .getSignatorySpsAuthentication();
+
+    String purposeString = spsCertificate.getSpsExchangedDocument().getSignatorySpsAuthentication()
+        .stream()
+        .flatMap(spsAuthentication -> spsAuthentication.getIncludedSpsClause().stream())
+        .filter(clause -> "PURPOSE".equals(clause.getId().getValue()))
+        .findAny()
+        .map(clause -> clause.getContent().get(0).getValue())
+        .orElse(null);
+    if (TRANSHIPMENT.toString().equals(purposeString)) {
+      return Purpose.builder()
+          .purposeGroup(TRANSHIPMENT_TO)
+          .finalBIP(retrieveFinalBip(spsCertificate))
+          .build();
+    } else {
+      for (SpsAuthenticationType spsAuthenticationType : spsAuthenticationTypeList) {
+        Purpose purpose = purposeMapper.map(spsAuthenticationType);
```

Comment:
I think I'm leaning towards not using the common PurposeMapper here, it's technically incorrect as it'll allow you to map purposes that aren't allowed in CHEDD. Try this instead:

```java
} else if (INTERNAL_MARKET.toString().equals(purposeString)) {
  return Purpose.builder()
      .purposeGroup(IMPORT)
      .build();
} else {
  throw new NotificationMapperException("Unable to map to the Purpose");
}
```

Tiny bit of duplicate from the common mapper but overall better as you can remove the dependency this class has on `PurposeMapper`